### ISSUE TITLE
[Self-managed][Agents]: 7.14.0 Snapshot Fleet Server agent went offline on triggering Agent Upgrade from Kibana UI.

### ISSUE THREAD

**Issue by amolnater-qasource** [elastic/beats#27453]
**Kibana version:** 7.15.0 Snapshot Kibana self-managed environment

**Host OS and Browser version:** AWS Windows Server 2019, All

**Build details:**
```
Build: 43546
Commit: 81e86207a4e276826fb0744e78239370f448c1ef
Artifact Link: https://snapshots.elastic.co/7.14.0-1738ccc6/downloads/beats/elastic-agent/elastic-agent-7.14.0-SNAPSHOT-windows-x86_64.zip
```

**Preconditions:**
1. 7.15.0 Snapshot Kibana self-managed environment should be available.
2. A Fleet Server agent must be installed with Default Fleet Server policy having Fleet Server integration.

**Steps to reproduce:**
1. Login to Kibana environment.
2. Navigate to Agents tab and Click Upgrade agent from Kibana UI.
3. Observe agent went offline.

**Expected Result:**
7.14.0 Snapshot Fleet Server agent should upgrade to 7.15.0 Snapshot on triggering Upgrade from Kibana UI.

**Logs:**
[logs.zip](https://github.com/elastic/beats/files/7005371/logs.zip)

**Default Fleet Server Policy:**
[elastic-agent.zip](https://github.com/elastic/beats/files/7005377/elastic-agent.zip)

**Screenshot:**
![1](https://user-images.githubusercontent.com/77374876/129855559-aa8775b8-4739-43b0-964c-a4c876d26c00.PNG)

**Comment by amolnater-qasource** [comment]
@manishgupta-qasource Please review.

**Comment by manishgupta-qasource** [comment]
Reviewed & assigned to @andresrc 

CC: @EricDavisX

**Comment by EricDavisX** [comment]
@blakerouse I tentatively assigned this to you, if you prefer to pull in another resource can you recommend and tag them please?

**Comment by EricDavisX** [comment]
@amolnater-qasource if Blake doesn't get to it today, can you try to test the latest 7.15 snapshot and if it works, the 7.14.1 release (with as close to same deploy infrastructure) to see if it works there?  We'd like to be able to close this if we are confident it shows as working now.  It isn't too late for a 7.15 fix, btw.

**Comment by blakerouse** [comment]
I was able to test this on Windows 10, upgrading from 7.14.0-SNAPSHOT to 7.15.0-SNAPSHOT and I am seeing the same behavior. There is something not working correctly here, looking into why.

**Comment by blakerouse** [comment]
Looks like Fleet Server starts but is unable to server web requests.

```
{"log.level":"error","service.name":"fleet-server","service.name":"fleet-server","message":"http: panic serving [fe80::d953:a2b5:5012:2646%Ethernet]:53073: runtime error: invalid memory address or nil pointer dereference\ngoroutine 618 [running]:\nnet/http.(*conn).serve.func1(0xc000360000)\n\t/var/lib/jenkins/.gvm/versions/go1.16.5.linux.amd64/src/net/http/server.go:1824 +0x153\npanic(0x1541aa0, 0x1ed3a70)\n\t/var/lib/jenkins/.gvm/versions/go1.16.5.linux.amd64/src/runtime/panic.go:971 +0x49a\ngithub.com/elastic/fleet-server/v7/cmd/fleet.(*CheckinT)._handleCheckin(0xc0005f4880, 0x17c9530, 0xc000290240, 0x1, 0x0, 0x0, 0xc000519200, 0x82, 0x1f4, 0xc000290250, ...)\n\t/var/lib/jenkins/workspace/elastic+unified-release+master+snapshot-standalone-build-fleet-server-7.15/cd/release/release-manager/build/projects/gopath/src/github.com/elastic/fleet-server/cmd/fleet/handleCheckin.go:161 +0x1b2\ngithub.com/elastic/fleet-server/v7/cmd/fleet.Router.handleCheckin(0x17e25e8, 0xc00012a200, 0x0, 0x0, 0xc0005f4880, 0xc0001a62d0, 0xc0005d8870, 0xc0000d8f90, 0x17c81d0, 0xc000374000, ...)\n\t/var/lib/jenkins/workspace/elastic+unified-release+master+snapshot-standalone-build-fleet-server-7.15/cd/release/release-manager/build/projects/gopath/src/github.com/elastic/fleet-server/cmd/fleet/handleCheckin.go:63 +0x4f5\ngithub.com/elastic/fleet-server/v7/internal/pkg/logger.HttpHandler.func1(0x17cf020, 0xc0001f09a0, 0xc000546800, 0xc0005ca020, 0x1, 0x1)\n\t/var/lib/jenkins/workspace/elastic+unified-release+master+snapshot-standalone-build-fleet-server-7.15/cd/release/release-manager/build/projects/gopath/src/github.com/elastic/fleet-server/internal/pkg/logger/http.go:112 +0xcf\ngithub.com/julienschmidt/httprouter.(*Router).ServeHTTP(0xc0001a0b40, 0x17cf020, 0xc0001f09a0, 0xc000546800)\n\t/var/lib/jenkins/workspace/elastic+unified-release+master+snapshot-standalone-build-fleet-server-7.15/cd/release/release-manager/build/projects/gopath/pkg/mod/github.com/julienschmidt/httprouter@v1.3.0/router.go:387 +0xc7e\nnet/http.serverHandler.ServeHTTP(0xc0008781c0, 0x17cf020, 0xc0001f09a0, 0xc000546800)\n\t/var/lib/jenkins/.gvm/versions/go1.16.5.linux.amd64/src/net/http/server.go:2887 +0xaa\nnet/http.(*conn).serve(0xc000360000, 0x17d0b20, 0xc0001981c0)\n\t/var/lib/jenkins/.gvm/versions/go1.16.5.linux.amd64/src/net/http/server.go:1952 +0x8cd\ncreated by net/http.(*Server).Serve\n\t/var/lib/jenkins/.gvm/versions/go1.16.5.linux.amd64/src/net/http/server.go:3013 +0x3b8\n","@timestamp":"2021-08-31T20:03:01.6Z"}
{"log.level":"error","service.name":"fleet-server","service.name":"fleet-server","message":"http: panic serving 10.0.2.15:58332: runtime error: invalid memory address or nil pointer dereference\ngoroutine 1012 [running]:\nnet/http.(*conn).serve.func1(0xc000360000)\n\t/var/lib/jenkins/.gvm/versions/go1.16.5.linux.amd64/src/net/http/server.go:1824 +0x153\npanic(0x1541aa0, 0x1ed3a70)\n\t/var/lib/jenkins/.gvm/versions/go1.16.5.linux.amd64/src/runtime/panic.go:971 +0x49a\ngithub.com/elastic/fleet-server/v7/cmd/fleet.(*CheckinT)._handleCheckin(0xc0005f4880, 0x17c9530, 0xc000290240, 0x1, 0x0, 0x0, 0xc000a7e200, 0x82, 0x1f4, 0xc000290250, ...)\n\t/var/lib/jenkins/workspace/elastic+unified-release+master+snapshot-standalone-build-fleet-server-7.15/cd/release/release-manager/build/projects/gopath/src/github.com/elastic/fleet-server/cmd/fleet/handleCheckin.go:161 +0x1b2\ngithub.com/elastic/fleet-server/v7/cmd/fleet.Router.handleCheckin(0x17e25e8, 0xc00012a200, 0x0, 0x0, 0xc0005f4880, 0xc0001a62d0, 0xc0005d8870, 0xc0000d8f90, 0x17c81d0, 0xc000374000, ...)\n\t/var/lib/jenkins/workspace/elastic+unified-release+master+snapshot-standalone-build-fleet-server-7.15/cd/release/release-manager/build/projects/gopath/src/github.com/elastic/fleet-server/cmd/fleet/handleCheckin.go:63 +0x4f5\ngithub.com/elastic/fleet-server/v7/internal/pkg/logger.HttpHandler.func1(0x17cf020, 0xc0001f0540, 0xc0006a0100, 0xc0000cc160, 0x1, 0x1)\n\t/var/lib/jenkins/workspace/elastic+unified-release+master+snapshot-standalone-build-fleet-server-7.15/cd/release/release-manager/build/projects/gopath/src/github.com/elastic/fleet-server/internal/pkg/logger/http.go:112 +0xcf\ngithub.com/julienschmidt/httprouter.(*Router).ServeHTTP(0xc0001a0b40, 0x17cf020, 0xc0001f0540, 0xc0006a0100)\n\t/var/lib/jenkins/workspace/elastic+unified-release+master+snapshot-standalone-build-fleet-server-7.15/cd/release/release-manager/build/projects/gopath/pkg/mod/github.com/julienschmidt/httprouter@v1.3.0/router.go:387 +0xc7e\nnet/http.serverHandler.ServeHTTP(0xc0008781c0, 0x17cf020, 0xc0001f0540, 0xc0006a0100)\n\t/var/lib/jenkins/.gvm/versions/go1.16.5.linux.amd64/src/net/http/server.go:2887 +0xaa\nnet/http.(*conn).serve(0xc000360000, 0x17d0b20, 0xc0005cc6c0)\n\t/var/lib/jenkins/.gvm/versions/go1.16.5.linux.amd64/src/net/http/server.go:1952 +0x8cd\ncreated by net/http.(*Server).Serve\n\t/var/lib/jenkins/.gvm/versions/go1.16.5.linux.amd64/src/net/http/server.go:3013 +0x3b8\n","@timestamp":"2021-08-31T20:10:23.781Z"}
```

**Comment by blakerouse** [comment]
I found the issue and filed it in Fleet Server here https://github.com/elastic/fleet-server/issues/682

**Comment by blakerouse** [comment]
This is closed with https://github.com/elastic/fleet-server/pull/683. I would re-test with a new snapshot once available or wait for 7.15.0 BC4.

**Comment by amolnater-qasource** [comment]
Hi @blakerouse 
Thanks for the update, we will re-test this issue too once new 7.15.0 Snapshot or 7.15.0 BC-4 will be available and will share our feedback on the same.

**Comment by amolnater-qasource** [comment]
Hi @EricDavisX 
We have revalidated this issue on 7.15.0 self-managed Snapshot and found it **fixed** now.
We installed 7.14.1 released agent using Default Fleet Server policy having Fleet Server integration and upgraded it to 7.15.0 Snapshot agent.

**Observations:**
- 7.14.1 released fleet server agent is successfully upgraded to 7.15.0 Snapshot.

**Build details:**
Build: 43916
Commit: 0de925293a62f83e548822f40ac1166666c3c415
Artifact Link: https://snapshots.elastic.co/7.15.0-b7f545f0/downloads/kibana/kibana-7.15.0-SNAPSHOT-windows-x86_64.zip

Thanks
